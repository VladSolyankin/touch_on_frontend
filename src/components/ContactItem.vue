<script setup lang="ts">
import Button from "../ui/Button.vue";
import { defineProps, defineEmits } from "vue";
import { Contact } from "../types";

const props = defineProps<{
  contact: Contact;
}>();

const emit = defineEmits(["delete", "edit"]);

const remove = () => {
  emit("delete", props.contact.id!);
};

const edit = () => {
  emit("edit", props.contact);
};
</script>

<template>
  <transition name="fade">
    <div class="contact-item">
      <span>{{ contact.name }}</span>
      <span>{{ contact.phone }}</span>
      <span>{{ contact.email }}</span>
      <Button @click="edit">Редактировать</Button>
      <Button @click="remove">Удалить</Button>
    </div>
  </transition>
</template>

<style>
.contact-item {
  display: flex;
  gap: 0.5rem;
  flex-direction: column;
}

button {
  margin-top: 0.5rem;
  padding: 0.5rem 1rem;
  border: 1px solid #ccc;
  border-radius: 4px;
  background-color: white;
  color: black;
  font-size: 1rem;
  font-family: inherit;
  cursor: pointer;

  transition: background-color 0.2s ease-in-out, color 0.2s ease-in-out;

  &:hover {
    background-color: black;
    color: white;
  }
}

.fade-enter-active,
.fade-leave-active {
  transition: opacity 0.5s ease;
}

.fade-enter-from,
.fade-leave-to {
  opacity: 0;
}
</style>
