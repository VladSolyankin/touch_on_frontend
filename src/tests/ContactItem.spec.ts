import { mount } from "@vue/test-utils";
import { describe, it, expect } from "vitest";
import ContactItem from "../components/ContactItem.vue";

describe("Тесты для ContactItem.vue", () => {
  it("Верный рендер контакта", () => {
    const contact = {
      id: 1,
      name: "John Doe",
      phone: "+71234567890",
      email: "john@example.com",
    };
    const wrapper = mount(ContactItem, {
      props: { contact },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Верный рендер при редактировании", async () => {
    const contact = {
      id: 1,
      name: "John Doe",
      phone: "+71234567890",
      email: "john@example.com",
    };
    const wrapper = mount(ContactItem, {
      props: { contact },
    });

    await wrapper.find("button").trigger("click");

    const emitted = wrapper.emitted("edit")!;
    expect(emitted).toHaveLength(1);
    expect(emitted[0][0]).toEqual(contact);
  });

  it("Верный рендер при удалении", async () => {
    const contact = {
      id: 1,
      name: "John Doe",
      phone: "+71234567890",
      email: "john@example.com",
    };
    const wrapper = mount(ContactItem, {
      props: { contact },
    });

    await wrapper.findAll("button")[1].trigger("click");

    const emitted = wrapper.emitted("delete")!;
    expect(emitted).toHaveLength(1);
    expect(emitted[0][0]).toBe(contact.id);
  });
});
