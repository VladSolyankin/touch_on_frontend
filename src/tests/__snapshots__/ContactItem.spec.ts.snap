// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Тесты для ContactItem.vue > Верный рендер контакта 1`] = `"<div data-v-f016766c="" class="contact-item"><span data-v-f016766c="">John Doe</span><span data-v-f016766c="">+71234567890</span><span data-v-f016766c="">john@example.com</span><button data-v-f016766c="">Редактировать</button><button data-v-f016766c="">Удалить</button></div>"`;
